jms/org.wso2.carbon.transport.jms/src/test/java/org/wso2/carbon/transport/jms/test/util/JMSServer.java

```diff
@@ -0,0 +1,131 @@
+import javax.jms.QueueConnection;
+import javax.jms.QueueSession;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+import javax.jms.TopicConnection;
+import javax.jms.TopicSession;
+
+/**
+ * A simple jms server using Activemq embedded broker.
+ */
+public class JMSServer {
+    private Logger logger = LoggerFactory.getLogger(JMSServer.class);
+    private ConnectionFactory connectionFactory;
+
+    /**
+     * To start the embedded activemq server.
+     */
+    public void startServer() {
+        connectionFactory = new ActiveMQConnectionFactory(JMSTestConstants.ACTIVEMQ_PROVIDER_URL);
+    }
+
+    /**
+     * To publish the messages to a queue.
+     *
+     * @throws JMSException         JMS Exception
+     * @throws InterruptedException Interrupted exception while waiting in between messages
+     */
+    public void publishMessagesToQueue(String queueName) throws JMSException, InterruptedException {
+        QueueConnection queueConn = (QueueConnection) connectionFactory.createConnection();
+        queueConn.start();
+        QueueSession queueSession = queueConn.createQueueSession(false, Session.AUTO_ACKNOWLEDGE);
+        Destination destination = queueSession.createQueue(queueName);
+        MessageProducer queueSender = queueSession.createProducer(destination);
+        queueSender.setDeliveryMode(DeliveryMode.NON_PERSISTENT);
+        for (int index = 0; index < 10; index++) {
+            String queueText = "Queue Message : " + (index + 1);
+            TextMessage queueMessage = queueSession.createTextMessage(queueText);
+            queueSender.send(queueMessage);
+            Thread.sleep(1000);
+            logger.info("Publishing " + queueText + " to queue " + queueName);
+        }
+        queueConn.close();
+        queueSession.close();
+        queueSender.close();
+    }
+
+    /**
+     * To publish the messages to a queue.
+     *
+     * @throws JMSException         JMS Exception
+     * @throws InterruptedException Interrupted exception while waiting in between messages
+     */
+    public void publishMessagesToTopic(String topicName) throws JMSException, InterruptedException {
+        TopicConnection topicConnection = (TopicConnection) connectionFactory.createConnection();
+        topicConnection.start();
+        TopicSession topicSession = topicConnection.createTopicSession(false, Session.AUTO_ACKNOWLEDGE);
+        Destination destination = topicSession.createTopic(topicName);
+        MessageProducer topicSender = topicSession.createProducer(destination);
+        topicSender.setDeliveryMode(DeliveryMode.NON_PERSISTENT);
+        for (int index = 0; index < 10; index++) {
+            String topicText = "Topic Message : " + (index + 1);
+            TextMessage topicMessage = topicSession.createTextMessage(topicText);
+            topicSender.send(topicMessage);
+            logger.info("Publishing " + topicText + " to topic " + topicName);
+            Thread.sleep(1000);
```

Comment:
Thread sleep with a constant value. Use something similar to RateLimitter in Guava

Reply:
This was considered as a future improvement.